### PR TITLE
support show variables where syntax

### LINKED ISSUE #199: support `show variables where` syntax
2015/09/19 21:39:04 session.go:235: [error] Syntax error: /* mysql-connector-java-5.1.35 ( Revision: 5fb9c5849535c13917c2cf9baaece6ef9693ef27 ) */SHOW VARIABLES WHERE Variable_name ='language' OR Variable_name = 'net_write_timeout' OR Variable_name = 'interactive_timeout' OR Variable_name = 'wait_timeout' OR Variable_name = 'character_set_client' OR Variable_name = 'character_set_connection' OR Variable_name = 'character_set' OR Variable_name = 'character_set_server' OR Variable_name = 'tx_isolation' OR Variable_name = 'transaction_isolation' OR Variable_name = 'character_set_results' OR Variable_name = 'timezone' OR Variable_name = 'time_zone' OR Variable_name = 'system_time_zone' OR Variable_name = 'lower_case_table_names' OR Variable_name = 'max_allowed_packet' OR Variable_name = 'net_buffer_length' OR Variable_name = 'sql_mode' OR Variable_name = 'query_cache_type' OR Variable_name = 'query_cache_size' OR Variable_name = 'license' OR Variable_name = 'init_connect'

### PR DESCRIPTION
**Pull request by siddontang**
try to fix #199 

@shenli

### CODE REVIEW

**Comment by coocood** [comment]
I also noticed that `WhereClauseOptional` returns `WhereRset` but all statements don't use that rset, but only use the expression.

**Comment by ngaut** [comment]
Please add some tests.

**Comment by siddontang** [comment]
@coocood

WhereClauseOptional returns a WhereRset, but I just want to use an expression to keep thing simple. 
But if your following pr merged, maybe we can use it.

**Comment by ngaut** [comment]
Also you need to add some tests to parser.

**Comment by siddontang** [comment]
PTAL @ngaut @coocood

File: tidb_test.go [hunk 1]
```diff
@@ -771,6 +771,18 @@ func (s *testSessionSuite) TestSubQuery(c *C) {
...
 
+func (s *testSessionSuite) TestShow(c *C) {
+	store := newStore(c, s.dbName)
+	se := newSession(c, store, s.dbName)
+
+	mustExecSQL(c, se, "set global autocommit=1")
+	r := mustExecSQL(c, se, "show variables where variable_name = 'autocommit'")
+	row, err := r.FirstRow()
+	c.Assert(err, IsNil)
+	match(c, row, "autocommit", 1)
```
Comment (ngaut) [inline tidb_test.go]: remove empty line.

**Comment by ngaut** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**